### PR TITLE
feat: add consistent alert modal #2154

### LINKED ISSUE #2154: [feature request] Consistent use of alert/modal
### Description of issue

There is an inconsistent use of modals across the website which isn't great from a user experience perspective. An example of this is that deleting comments for research articles and howtos uses `window.confirm`  Modals should be used to replace these window alerts and confirmations.

**Window.confirm**
image

**Modal**
image

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

1. Adds common confirm modal component
2. Replaces comment deletion window.confirm in how to and research pages with new confirm modal
3. Replaces contact link delete confirmation on settings page with new confirm modal
4. Adds / updates cypress tests for pages (tests are to the same level of the pre-existing ones but need improving, added a todo for myself)

## Git Issues

Closes #2154 

## Screenshots/Videos

image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: packages/cypress/src/integration/howto/comments.spec.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          .get(`[data-cy="CommentItem: delete button"]`)
+          .click()
+
+        cy.get('[data-cy="Confirm.modal: Confirm"]').should('exist')
+
+        // TODO: check db to make sure comment deleted
+      })
+    })
+  })
+})
```
Comment (AlfonsoGhislieri) [inline packages/cypress/src/integration/howto/comments.spec.ts:79]: **praise**: Good idea to get the comments into a separate spec file

File: packages/cypress/src/integration/howto/read.spec.ts [hunk 2]
```diff
@@ -151,60 +151,62 @@ describe('[How To]', () => {
...
+    //     })
+
+    //     it('[Logged in user can edit the comment they have added]', () => {
+    //       cy.login('howto_reader@test.com', 'test1234')
+    //       cy.visit(specificHowtoUrl)
+
+    //       cy.get('[data-cy="CommentItem: edit button"]').should('exist')
+    //     })
+    //   })
+    // })
```
Comment (AlfonsoGhislieri) [inline packages/cypress/src/integration/howto/read.spec.ts:209]: You delete these lines, I agree with your changes to move the comment tests to a separate spec file. It's in line with how the research comment testing is handled.

**Review by AlfonsoGhislieri — CHANGES_REQUESTED** [review]

**Comment by AlfonsoGhislieri** [comment]
Thanks a lot @iSCJT, looks really good to me, great addition to have a new component! 🤩

File: packages/components/src/ConfirmModal/ConfirmModal.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Box, Flex, Text } from 'theme-ui'
+import { Button, Modal } from '..'
+
+export const ConfirmModal: React.FC<{
```
[the remark this answers is not in the document]
Comment (iSCJT) [reply packages/components/src/ConfirmModal/ConfirmModal.tsx:4]: I'll refactor, I wouldn't normally use but I checked how another component was implemented and went for consistency.  I'll add myself a chore to refactor the others as well.

File: packages/cypress/src/integration/howto/comments.spec.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        cy.get('[data-cy="comment-text"]').should('contain.text', commentText)
+
+        cy.get('[data-cy="comment-text"]')
+          .get(`[data-cy="CommentItem: delete button"]`)
+          .click()
+
+        cy.get('[data-cy="Confirm.modal: Confirm"]').should('exist')
+
+        // TODO: check db to make sure comment deleted
```
[the remark this answers is not in the document]
Comment (iSCJT) [reply packages/cypress/src/integration/howto/comments.spec.ts:75]: Link doesn't work but i'll find it in the codebase.  Thanks for highlighting!  I'll clear the TODO's

File: packages/components/src/ConfirmModal/ConfirmModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (AlfonsoGhislieri) [reply packages/components/src/ConfirmModal/ConfirmModal.tsx:4]: Yeah I do think there is a little inconsistency from back when it still was reccomended

File: packages/cypress/src/integration/howto/comments.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlfonsoGhislieri) [reply packages/cypress/src/integration/howto/comments.spec.ts:75]: Sorry, this is the one [here](https://github.com/ONEARMY/community-platform/blob/6947c477525c6b4d7ea3c222e144134ecc4d99f7/src/stores/Howto/howto.store.test.ts)

File: packages/components/src/ConfirmModal/ConfirmModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (thisislawatts) [reply packages/components/src/ConfirmModal/ConfirmModal.tsx:4]: >  I'll add myself a chore to refactor the others as well.
Perhaps useful to raise this is as a GitHub Issue, open it up for anyone looking to make contributions to the project?

File: packages/cypress/src/integration/howto/comments.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (iSCJT) [reply packages/cypress/src/integration/howto/comments.spec.ts:75]: thanks @AlfonsoGhislieri

File: packages/components/src/ConfirmModal/ConfirmModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (iSCJT) [reply packages/components/src/ConfirmModal/ConfirmModal.tsx]: @thisislawatts good idea, I've added #2182

**Review by AlfonsoGhislieri — APPROVED** [review]

**Comment by davehakkens** [comment]
The branch seems to have some conflicts? Can't merge yet :|
```
 src/pages/Howto/Content/Howto/HowToComments/HowToComments.tsx
src/pages/Research/Content/ResearchComments/ResearchComments.tsx 
```

**Review by davehakkens — APPROVED** [review]

**Pull request merged**